packages/client/src/features/search-palette/search-palette.ts

```diff
@@ -0,0 +1,166 @@
+/********************************************************************************
+ * Copyright (c) 2023-2025 Business Informatics Group (TU Wien) and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { GModelElement, GModelRoot, GNode, hasProperty, LabeledAction, SelectAllAction, toArray, TYPES } from '@eclipse-glsp/sprotty';
+import { inject, injectable } from 'inversify';
+import { debounce, isEqual } from 'lodash';
+import {
+    AutocompleteSuggestionProviderContext,
+    type AutocompleteSuggestion,
+    type AutocompleteSuggestionRegistry
+} from '../../base/auto-complete/autocomplete-suggestion-provider';
+import { BaseAutocompletePalette } from '../../base/auto-complete/base-autocomplete-palette';
+import { applyCssClasses, deleteCssClasses } from '../../base/feedback/css-feedback';
+import { messages } from '../../base/messages';
+import { GEdge } from '../../model';
+import { RepositionAction } from '../viewport/reposition';
+
+const CSS_SEARCH_HIDDEN = 'search-hidden';
+const CSS_SEARCH_HIGHLIGHTED = 'search-highlighted';
+
+export interface SearchAutocompleteSuggestion extends AutocompleteSuggestion {
+    element: GModelElement;
+}
+
+export namespace SearchAutocompleteSuggestion {
+    export function is(suggestion: AutocompleteSuggestion): suggestion is SearchAutocompleteSuggestion {
+        return hasProperty<SearchAutocompleteSuggestion>(suggestion, 'element');
```

Comment:
Since we already have a `AutocompleteSuggestion` as input you can simply check if element is defined:
```suggestion
        return suggestion.element !== undefined;
```